lib/internal/fs/promises.js

```diff
@@ -668,11 +669,17 @@ async function writeFile(path, data, options) {
     data = Buffer.from(data, options.encoding || 'utf8');
   }
 
+  validateAbortSignal(options.signal);
   if (path instanceof FileHandle)
     return writeFileHandle(path, data, options.signal);
 
+  if (options.signal?.aborted) {
+    throw lazyDOMException('The operation was aborted', 'AbortError');
+  }
+
   const fd = await open(path, flag, options.mode);
   if (options.signal?.aborted) {
+    await fd.close();
     throw lazyDOMException('The operation was aborted', 'AbortError');
   }
   return PromisePrototypeFinally(writeFileHandle(fd, data), fd.close);
```

Comment:
Shouldn't we throw the DOMException right? In case `fd.close()` is rejected or takes a long time to resolve, it would create a weird behavior.
```suggestion
  const fd = await open(path, flag, options.mode);
  try {
    if (options.signal?.aborted) {
      throw lazyDOMException('The operation was aborted', 'AbortError');
    }
    return await writeFileHandle(fd, data);
  } finally {
    await fd.close();
  }
```

Reply:
I don't mind changing to this, I just don't think that this would behave differently (as the finally happens before the promise resolves, and if you throw in the finally it would still throw).

Reply:
I've changed the code to match the above